boards/arm/samv7/common/Kconfig

```diff
@@ -0,0 +1,96 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfiglanguage.txt in the NuttX tools repository.
+#
+
+config BOARD_SAMV7_COMMON
+	bool "Board common logic"
+	default y
+	help
+		 Board common logic located in each board/common folder.
+
+if BOARD_SAMV7_COMMON
+
+config SAMV7_PROGMEM_OTA_PARTITION
+	bool
+	default n
+	select ARCH_RAMFUNCS
+	select BCH
+	select MTD
+	select MTD_BYTE_WRITE
+	select MTD_CONFIG
```

Comment:
please double check if `MTD_CONFIG` is mandatory. I think it can be removed from the list

Reply:
MTD_CONFIG enables MTD_CONFIG_ERASEDVALUE https://github.com/apache/incubator-nuttx/blob/51a2db6ffc4660346dddf6b577089d6f1dc9c504/drivers/mtd/Kconfig#L181

Reply:
Yeah, but why do we need `MTD_CONFIG_ERASEDVALUE`?

Reply:
This store the correct value of CONFIG_MCUBOOT_DEFAULT_FLASH_ERASE_STATE. I'll send an update later dropping it from this sequence. That is a remaining from a previous attempt to reorganize MCUboot. However, it seems more prudent first reorganize samv7 and then propose changes in a future PR.

Reply:
Yes, please explain. What I read about `MCUBOOT_DEFAULT_FLASH_ERASE_STATE` in `apps` repo is:
```
config MTD_CONFIG_ERASEDVALUE
	hex "Erased value of bytes on the MTD device"
	default 0xff
	---help---
		Specifies the value of the erased state of the MTD FLASH.  For
		most FLASH parts, this is 0xff, but could also be zero depending
		on the device.
```
And per my understanding has nothing to do with `MTD_CONFIG_ERASEDVALUE`.
`MTD_CONFIG` is:
```
config MTD_CONFIG
	bool "Enable Dev Config (MTD based) device"
	default n
	---help---
		Provides a /dev/config device for saving / restoring application
		configuration data to a standard MTD device or partition.
```
and it is used in `sam_at24config.c` for `at24` based example.
The MCUboot relies on `CONFIG_MTD_PROGMEM_ERASESTATE` that supplies flash erase value `ioctl(MTDIOC_ERASESTATE)`.

Reply:
I understand that MCUBOOT_DEFAULT_FLASH_ERASE_STATE should be dropped in favor to use the well defined Kconfig at NuttX, not at Apps, to initialize structures, as below:

https://github.com/mcu-tools/mcuboot/blob/e9c6b4d7af7598fbd915868cf89383ff854d2241/boot/nuttx/src/flash_map_backend/flash_map_backend.c#L85-L99

I almost sure now that I interpreted wrongly the use of MTD_CONFIG_ERASEDVALUE. The thing is that it is important define a way that serves for all MTD use cases. The CONFIG_MTD_PROGMEM_ERASESTATE won't work to replace MCUBOOT_DEFAULT_FLASH_ERASE_STATE. In my view this should be supplied by NuttX to any bootloader App instead rely on user to configure this.

Anyway, this won't be part of this PR as already stated earlier.

Reply:
Repeating again that I'm not talking about dependency of `CONFIG_MTD_PROGMEM_ERASESTATE` or `MCUBOOT_DEFAULT_FLASH_ERASE_STATE`. My initial comment is that `SAMV7_PROGMEM_OTA_PARTITION` should not select `MTD_CONFIG` because there is not dependency between this two options because MCUboot partition is located in MTD PROGMEM dev.
In case if MCUboot partition is located in MTD CONFIG dev, then we can have such dependency.

Reply:
Dropped MTD_CONFIG

Reply:
I will recheck on qmtech board and reply with my approve